sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncContainer.java

```diff
@@ -825,15 +846,19 @@ private <T> Mono<CosmosItemResponse<T>> createItemHelper(byte[] streamPayload,
                                                              Class<T> itemClass,
                                                              boolean isRetry) {
         this.setRequestHeaders(requestOptions);
+        CosmosItemSerializer effectiveItemSerializer =
+            cosmosEncryptionAsyncClient.getEffectiveItemSerializer(requestOptions.getCustomSerializer());
```

Comment:
do we need to set the default serializer in the request options?

Reply:
Why - we always take from RequestOptions or fallback to Client-wide settings.

Reply:
but in encryption, I thought the one we passed to azure-cosmos is always default serializer as the customized serialization happens in encryption?